### PR TITLE
[ADHOC] feat(sdk): mandatory knownSignatures for validation

### PR DESCRIPTION
**Pull request by sammccord**
🚨 Please review the [guidelines for contributing](https://github.com/boostxyz/boost-protocol/blob/main/.github/CONTRIBUTING.md) to this repository.

### Description

make `ValidationStepParams.knownSignatures` mandatory, remove dependency on @boostxyz/signatures from @boostxyz/sdk

also don't publish signatures and sdk together

💔 Thank you!

### CODE REVIEW

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  By using the `[ADHOC]` title prefix, you are bypassing best practice protections.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 4eac688b4efec4f7b136af2671df66569a05f84d
</p>

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  By using the `[ADHOC]` title prefix, you are bypassing best practice protections.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 167fdff9b7dd75814d08dfda06b9619f5ad8415e
</p>

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  By using the `[ADHOC]` title prefix, you are bypassing best practice protections.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 9bc2c79597c3017f70b1610eb6b9745305a16e3f
</p>

File: packages/sdk/src/Actions/EventAction.ts [hunk 1]
```diff
@@ -227,14 +217,14 @@ export interface ActionStep {
  * Parameters for validating an action step.
  *
  * @typedef {Object} ValidateActionStepParams
- * @property {Record<Hex, AbiEvent | AbiFunction>} [knownSignatures] - Optional record of known events, keyed by 32 byte selectors.
+ * @property {Record<Hex, AbiEvent | AbiFunction>} [knownSignatures] - Record of known events, keyed by 32 byte selectors. You can use [@boostxyz/signaturs](https://www.npmjs.com/package/@boostxyz/signatures) to assemble this parameter.
```
Comment (mmackz) [inline packages/sdk/src/Actions/EventAction.ts:220]: @boostxyz/signaturs -> @boostxyz/signatures

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  By using the `[ADHOC]` title prefix, you are bypassing best practice protections.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 20ac4914e9f61a236411b480b8fb29e61a826b89
</p>

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  By using the `[ADHOC]` title prefix, you are bypassing best practice protections.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 20ac4914e9f61a236411b480b8fb29e61a826b89
</p>

**Pull request merged**